enterprise/server/remote_execution/execution_server/execution_server_test.go

```diff
@@ -0,0 +1,144 @@
+package execution_server_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"gorm.io/gorm"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/execution_server"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/testutil/testredis"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/util/redisutil"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/testenv"
+
+	"github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+type schedulerServerMock struct {
+	interfaces.SchedulerService
+}
+
+func (s *schedulerServerMock) CancelTask(ctx context.Context, taskID string) (bool, error) {
+	return true, nil
+}
+
+func setupEnv(t *testing.T) *testenv.TestEnv {
+	env := testenv.GetTestEnv(t)
+
+	redisTarget := testredis.Start(t).Target
+	rdb := redis.NewClient(redisutil.TargetToOptions(redisTarget))
+	env.SetRemoteExecutionRedisClient(rdb)
+	env.SetRemoteExecutionRedisPubSubClient(rdb)
+
+	scheduler := &schedulerServerMock{}
+	env.SetSchedulerService(scheduler)
+
+	s, err := execution_server.NewExecutionServer(env)
+	require.NoError(t, err)
+	env.SetRemoteExecutionService(s)
+
+	return env
+}
+
+func createExecution(t *testing.T, db *gorm.DB, execution *tables.Execution) {
+	err := db.Create(execution).Error
+	require.NoError(t, err)
+}
+
+func TestCancel(t *testing.T) {
+	env := setupEnv(t)
+	ctx := context.Background()
+	s := env.GetRemoteExecutionService()
+
+	// Create Execution rows to be canceled
+	db := env.GetDBHandle().DB(ctx)
+	testUUID, err := uuid.NewRandom()
+	require.NoError(t, err)
+	testInvocationID := testUUID.String()
+
+	executionID := "blobs/1111111111111111111111111111111111111111111111111111111111111111/100"
+	execution := &tables.Execution{
+		ExecutionID:  executionID,
+		InvocationID: testInvocationID,
+		Stage:        int64(remote_execution.ExecutionStage_EXECUTING),
+	}
+	createExecution(t, db, execution)
+
+	numCanceled, err := s.Cancel(ctx, testInvocationID)
```

Comment:
You could have your fake scheduler server count the # of cancellations and avoid the need to return the # of cancelled executions for the sake of the test.

Reply:
Good call -done